### PR TITLE
Add the ability to pass args to the rsh/ssh command line

### PR DESCRIPTION
**Pull request by rhc54**
Per request from Andy Rieb, add ability to pass PATH and LD_LIBRARY_PATH elements to ssh command

(cherry picked from commit open-mpi/ompi@6f227f8564d68e40437cadafe814f59f22ae6d1f)
(cherry picked from commit open-mpi/ompi@278324c52a6937b6371610727bf02511945b47d3)

### CODE REVIEW

**Comment by mellanox-github** [comment]
- Coverity found 1473 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/all_222/output/errors/index.html
- Coverity found 23 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/oshmem_222/output/errors/index.html
- Coverity found 7 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/fca_222/output/errors/index.html <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/222/
Test PASSed.

**Comment by rhc54** [comment]
Needs fix - coming over weekend

**Comment by mellanox-github** [comment]
- Coverity found 1473 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/all_225/output/errors/index.html
- Coverity found 23 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/oshmem_225/output/errors/index.html
- Coverity found 7 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/fca_225/output/errors/index.html <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/225/
Test PASSed.

**Comment by mellanox-github** [comment]
- Coverity found 1468 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/all_229/output/errors/index.html
- Coverity found 23 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/oshmem_229/output/errors/index.html
- Coverity found 7 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/fca_229/output/errors/index.html <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/229/
Test PASSed.

File: orte/mca/plm/rsh/plm_rsh_module.c [hunk 1]
```diff
@@ -433,6 +446,19 @@ static int setup_launch(int *argcptr, char ***argvptr,
...
     }
     opal_argv_free(orted_argv);  /* done with this */
+
+    /* if the user specified a library path to pass, set it up now */
+    param = opal_basename(opal_install_dirs.libdir);
+    if (NULL != mca_plm_rsh_component.pass_libpath) {
+        if (NULL != prefix_dir) {
+            asprintf(&lib_base, "%s:%s/%s", mca_plm_rsh_component.pass_libpath, prefix_dir, param);
+        } else {
+            asprintf(&lib_base, "%s:%s", mca_plm_rsh_component.pass_libpath, param);
```
Comment (jsquyres) [inline orte/mca/plm/rsh/plm_rsh_module.c]: If prefix_dir is NULL, shouldn't you add the entire opal_install_dirs.libdir (not just the param, which is the basename)?  It seems odd that all of the libdir is dropped except the basename -- the MCA param description doesn't describe this value this way:

Thinking about this more, the interaction of --prefix and this libdir param is very... odd.  I see how you got here, but it still feels... weird.

Want to chat on the phone about some options and/or rationale here?

Comment (rhc54) [inline orte/mca/plm/rsh/plm_rsh_module.c]: Sure - I'm available at 7am Pacific tomorrow, or right after the OMPI
telecon.

**Comment by rhc54** [comment]
@jsquyres do these revised comments resolve your concerns?

**Comment by mellanox-github** [comment]
- Coverity found 1468 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/all_235/output/errors/index.html
- Coverity found 23 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/oshmem_235/output/errors/index.html
- Coverity found 7 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/fca_235/output/errors/index.html <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/235/
Test PASSed.

**Comment by jsquyres** [comment]
:+1:

**Pull request merged**